components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceAgentServiceTest.java

```diff
@@ -0,0 +1,655 @@
+import org.wso2.carbon.utils.CarbonUtils;
+
+import javax.cache.CacheManager;
+import javax.ws.rs.core.Response;
+import java.rmi.RemoteException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+/**
+ * This class holds the unit tests for the class {@link DeviceAgentServiceImpl}
+ */
+@PowerMockIgnore("javax.ws.rs.*")
+@SuppressStaticInitializationFor({"org.wso2.carbon.device.mgt.jaxrs.util.DeviceMgtAPIUtils",
+        "org.wso2.carbon.context.CarbonContext", "org.wso2.carbon.context.internal.CarbonContextDataHolder"})
+@PrepareForTest({DeviceMgtAPIUtils.class, DeviceManagementProviderService.class,
+        DeviceAccessAuthorizationService.class, EventStreamAdminServiceStub.class, PrivilegedCarbonContext.class,
+        CarbonContext.class, CarbonUtils.class})
+public class DeviceAgentServiceTest {
+
+    private static final Log log = LogFactory.getLog(DeviceTypeManagementAdminService.class);
+    private DeviceManagementProviderService deviceManagementProviderService;
+    private DeviceAgentService deviceAgentService;
+    private EventStreamAdminServiceStub eventStreamAdminServiceStub;
+    private PrivilegedCarbonContext privilegedCarbonContext;
+    private CarbonContext carbonContext;
+    private CacheManager cacheManager;
+    private DeviceAccessAuthorizationService deviceAccessAuthorizationService;
+    private static final String TEST_DEVICE_TYPE = "TEST-DEVICE-TYPE";
+    private static final String TEST_DEVICE_IDENTIFIER = "11222334455";
+    private static final String AUTHENTICATED_USER = "admin";
+    private static final String TENANT_DOMAIN = "carbon.super";
+    private static Device demoDevice;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeClass
+    public void init() {
+        log.info("Initializing DeviceAgent tests");
+        initMocks(this);
+        this.deviceManagementProviderService = Mockito
+                .mock(DeviceManagementProviderServiceImpl.class, Mockito.RETURNS_MOCKS);
+        this.deviceAgentService = new DeviceAgentServiceImpl();
+        this.deviceAccessAuthorizationService = Mockito.mock(DeviceAccessAuthorizationServiceImpl.class,
+                Mockito.RETURNS_MOCKS);
+        this.privilegedCarbonContext = Mockito.mock(PrivilegedCarbonContext.class, Mockito.RETURNS_MOCKS);
+        this.carbonContext = Mockito.mock(CarbonContext.class, Mockito.RETURNS_MOCKS);
+        this.eventStreamAdminServiceStub = Mockito.mock(EventStreamAdminServiceStub.class, Mockito.RETURNS_MOCKS);
+        demoDevice = DeviceMgtAPITestHelper.generateDummyDevice(TEST_DEVICE_TYPE, TEST_DEVICE_IDENTIFIER);
+        this.cacheManager = Mockito.mock(CacheManager.class, Mockito.RETURNS_MOCKS);
+    }
+
+    @Test(description = "Test device Enrollment when the device is null")
+    public void testEnrollDeviceWithDeviceIsNULL() {
```

Comment:
Shall we use the correct convention when naming the test methods

Reply:
This is fixed in the commit 
https://github.com/wso2/carbon-device-mgt/pull/1082/commits/1a995dde138fc934f14bf626ea0506061f04f39e